pennylane_lightning/src/tests/lightning_util_unittest.cpp

```diff
@@ -14,33 +14,47 @@
 #include "gtest/gtest.h"
 #include "../rework/Util.hpp"
 
+#include <tuple>
+
 namespace test_utils {
 
-TEST(exp2, fixed_example) {
-    std::vector<unsigned int> inputs = {1, 2, 5, 8};
-    std::vector<size_t> outputs = {2, 4, 32, 256};
+class Exp2TestFixture :public ::testing::TestWithParam<std::tuple<int, int>> {
+};
 
-    for (size_t i = 0; i < inputs.size(); i++) {
-        size_t result = Pennylane::exp2(inputs[i]);
-        EXPECT_TRUE(result == outputs[i]);
-    }
+TEST_P(Exp2TestFixture, CheckExp2Results) {
+    int input = std::get<0>(GetParam());
+    int expected = std::get<1>(GetParam());
```

Comment:
Do you get any warnings when using an `int` type? For me, it was giving a warning of comparing `int` and `size_t`.

Reply:
Yep, same here, warnings are emitted there.